### PR TITLE
Add Bubbleprof text to io recommendation

### PR DESCRIPTION
**Pull request by AlanSl**
Now that Clinic Bubbleprof is available, we can recommend it in Doctor. This is a starting point, I'm expecting some suggested versions.

New text as shown in the UI:

image

image

### CODE REVIEW

**Comment by mcollina** [comment]
Let's wait @dave anyway.

File: recommendations/io-readmore.md [hunk 1]
```diff
@@ -13,29 +13,20 @@ The profiled process has been observed is unusually idle under load, typically t
...
 To solve I/O issues we have to track down the asynchronous call(s) which are taking an
-abnormally long time to complete.
+abnormally long time to complete. I/O root cause analysis is mostly a reasoning exercise.
 
-I/O root cause analysis is mostly a reasoning exercises, or requires advanced knowledge and
-expertise with specialist Node.js logging flags and (very new) asynchronous tracking API's.
-
-At nearForm we care a lot about this problem and we are developing a new tool to make
-I/O debugging easier... stay tuned!
+It used to requires advanced knowledge and
```
Comment (davidmarkclements) [inline recommendations/io-readmore.md]: I disagree with this approach - I think the text should be more timeless - I don't think it should sound like marketing blurb here. The tone, imo, should reflect that of the event loop recommendation, namely "Bubbleprof is the tool to use in this situation"

File: recommendations/io-readmore.md [hunk 2]
```diff
@@ -13,29 +13,20 @@ The profiled process has been observed is unusually idle under load, typically t
...
-  - Another approach is write benchmarks specifically for the I/O touch points
-    - Be sure to duplicate the exact conditions as generated in the Node.js process
-- Run these measurements multiple times, look for unreasonably high response times
-- Once slow I/O touch points have been discovered a strategy for speeding up the I/O is required  
-
-**Advanced**: For more advanced, lower overhead, timing functionality, check out the experimental `perf_hooks` API  
-
-**Advanced:** An alternative to the approach outlined above is to make use of the experimental
-Node.js `async_hooks` API, in combination with a timer and stack trace generation.
+- Make sure you are aware of the application's I/O "touch points": that is, I/O to and from the Node.js process (e.g. databases, network requests, filesystem...)
```
Comment (davidmarkclements) [inline recommendations/io-readmore.md]: can we avoid changing pronoun to 2nd person, all of the recommendation are written in 1st person plural + 3rd person omniscient

File: recommendations/io-readmore.md [hunk 3]
```diff
@@ -13,29 +13,20 @@ The profiled process has been observed is unusually idle under load, typically t
...
+- Use `clinic bubbleprof` to create a diagram of the application's asynchronous flow.
+  - See `clinic bubbleprof --help` for how to generate the profile
+  - Visit https://clinicjs.org/bubbleprof/walkthrough for a guide on how to use and interpret this output
+
+- Explore the Bubbleprof diagram. Look for long lines and large circles, representing persistent delays, then drill down to reveal the lines of code responsible
```
Comment (davidmarkclements) [inline recommendations/io-readmore.md]: comma after circles should be removed

File: recommendations/io-readmore.md [hunk 4]
```diff
@@ -13,29 +13,20 @@ The profiled process has been observed is unusually idle under load, typically t
...
+- Pay particular attention to "userland" delays, originating from code in the profiled application itself.
+- Use your understanding of the application's I/O touch points to identify possible optimization targets, such as operations in series which could be executed in parallel, or slow operations that can be optimised externally (for example with caching or indexing). Some large processes will have good reasons for being almost constantly in the queue (for example, some server handlers)
```
Comment (davidmarkclements) [inline recommendations/io-readmore.md]: run on sentence - can you rephrase into shorter sentences, maybe avoid the parens aswell

**Comment by AlanSl** [comment]
@davidmarkclements Couldn't contact you today, presume you're in SF, we're merging with the above changes, ping me if there's anything more you want changed and I'll open a new PR for it
